### PR TITLE
Add region to all OSB API requests

### PR DESCRIPTION
**Pull request by mszostok**
**Description**

Changes proposed in this pull request:

- Add region to all OSB API requests.
- Add documentation about all available OSB API endpoints prefixes

**Related issue(s)**

- https://github.com/kyma-incubator/compass/issues/1166

### CODE REVIEW

File: components/kyma-environment-broker/internal/broker/instance_create.go [hunk 1]
```diff
@@ -74,12 +75,20 @@ func (b *ProvisionEndpoint) Provision(ctx context.Context, instanceID string, de
...
+
 	provisioningParameters := internal.ProvisioningParameters{
 		PlanID:     details.PlanID,
 		ServiceID:  details.ServiceID,
 		ErsContext: ersContext,
 		Parameters: parameters,
+		Region:     region,
 	}
+
 	logger.Infof("Starting provisioning runtime: Name=%s, GlobalAccountID=%s, SubAccountID=%s", parameters.Name, ersContext.GlobalAccountID, ersContext.SubAccountID)
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/broker/instance_create.go:92]: please, add a region to log (or there is a place where you are logging region but I didn'f find it)

**Comment by piotrmiskiewicz** [comment]
Everything looks fine except naming: currently we have a region also in ProvisioningParametersDTO (it is an Azure or GCP region). I'm thinking about a different name for the region you were introducing in the PR. The same name could be misleading. Maybe "BrokerRegion" or you have a better idea.

File: components/kyma-environment-broker/internal/broker/instance_create.go (hunk 1 above)
Comment (mszostok) [reply components/kyma-environment-broker/internal/broker/instance_create.go]: done :)

File: docs/kyma-environment-broker/02-02-broker-endpoints.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Kyma Environment Broker Endpoints
+
+Kyma Environment Broker (KEB) implements the [Open Service Broker API](https://github.com/openservicebrokerapi/servicebroker/blob/v2.14/profile.md#service-metadata) (OSB API). All the OSB API endpoints are served with the following prefixes: 
+
+| Prefix            | Description                                                                                                                                                                                                                                      |
+|-------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `/`               | Defines prefix for legacy endpoint secured by basic access authentication. EDP is configured with region which defaults to **broker.defaultRequestRegion** value specified in [values.yaml](./../../chart/compass/charts/kyma-environment-broker/values.yaml) file. |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/02-02-broker-endpoints.md:7]:
```suggestion
| `/`               | Defines a prefix for the legacy endpoint secured with the basic access authentication. EDP is configured with a region whose default value is specified under the **broker.defaultRequestRegion** parameter in the [`values.yaml`](./../../chart/compass/charts/kyma-environment-broker/values.yaml) file. |
```

File: docs/kyma-environment-broker/02-02-broker-endpoints.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+| `/{region}`       | Defines prefix for legacy endpoint secured by basic access authentication. EDP is configured with region value specified in request.                                                                                                                                |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/02-02-broker-endpoints.md:8]:
```suggestion
| `/{region}`       | Defines a prefix for the legacy endpoint secured with the basic access authentication. EDP is configured with the region value specified in the request.                                                                                                                                |
```

File: docs/kyma-environment-broker/02-02-broker-endpoints.md (hunk 2 above)
Comment (klaudiagrz) [reply docs/kyma-environment-broker/02-02-broker-endpoints.md:7]: Authentication or authorization?

File: docs/kyma-environment-broker/02-02-broker-endpoints.md (hunk 3 above)
Comment (klaudiagrz) [reply docs/kyma-environment-broker/02-02-broker-endpoints.md:8]: Authentication or authorization?

File: docs/kyma-environment-broker/02-02-broker-endpoints.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+| `/oauth`          | Defines prefix for endpoint secured by OAuth2 authorization. EDP is configured with region which defaults to **broker.defaultRequestRegion** value specified in [values.yaml](./../../chart/compass/charts/kyma-environment-broker/values.yaml) file.               |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/02-02-broker-endpoints.md:9]:
```suggestion
| `/oauth`          | Defines a prefix for the endpoint secured with the OAuth2 authorization. EDP is configured with a region whose default value is specified under the  **broker.defaultRequestRegion** parameter specified in the [`values.yaml`](./../../chart/compass/charts/kyma-environment-broker/values.yaml) file.               |
```

File: docs/kyma-environment-broker/02-02-broker-endpoints.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+| `/oauth/{region}` | Defines prefix for endpoint secured by OAuth2 authorization. EDP is configured with region value specified in request.                                                                                                                           |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/02-02-broker-endpoints.md:10]:
```suggestion
| `/oauth/{region}` | Defines a prefix for the endpoint secured with the OAuth2 authorization. EDP is configured with the region value specified in the request.                                                                                                                           |
```

File: docs/kyma-environment-broker/02-02-broker-endpoints.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+
+> **NOTE:** The KEB does not implement the OSB API update operation.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/02-02-broker-endpoints.md:13]:
```suggestion
> **NOTE:** KEB does not implement the OSB API update operation.
```

File: docs/kyma-environment-broker/02-02-broker-endpoints.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+Beside OSB API endpoints KEB exposes the REST `/info/runtimes` endpoint that provides information about all created Runtimes, both succeeded and failed. This endpoint is secured with the OAuth2 authorization.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/02-02-broker-endpoints.md:15]:
```suggestion
Besides OSB API endpoints, KEB exposes the REST `/info/runtimes` endpoint that provides information about all created Runtimes, both succeeded and failed. This endpoint is secured with the OAuth2 authorization.
```

File: docs/kyma-environment-broker/02-02-broker-endpoints.md (hunk 2 above)
Comment (mszostok) [reply docs/kyma-environment-broker/02-02-broker-endpoints.md:7]: basic is a `authentication` :)

File: docs/kyma-environment-broker/02-02-broker-endpoints.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+| `/`               | Defines a prefix for the legacy endpoint secured with the basic access authentication. EDP is configured with a region whose default value is specified under the **broker.defaultRequestRegion** parameter in the [`values.yaml`](./../../chart/compass/charts/kyma-environment-broker/values.yaml) file. |
+| `/{region}`       | Defines a prefix for the legacy endpoint secured with the basic access authentication. EDP is configured with the region value specified in the request.                                                                                                                                |
+| `/oauth`          | Defines a prefix for the endpoint secured with the OAuth2 authorization. EDP is configured with a region whose default value is specified under the  **broker.defaultRequestRegion** parameter specified in the [`values.yaml`](./../../chart/compass/charts/kyma-environment-broker/values.yaml) file.               |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/02-02-broker-endpoints.md:9]:
```suggestion
| `/oauth`          | Defines a prefix for the endpoint secured with the OAuth2 authorization. EDP is configured with a region whose default value is specified under the **broker.defaultRequestRegion** parameter in the [`values.yaml`](./../../chart/compass/charts/kyma-environment-broker/values.yaml) file.               |
```

**Pull request merged**